static/js/components/TopSources.jsx

```diff
@@ -1,64 +1,228 @@
-import React from "react";
-import { useSelector } from "react-redux";
+import React, { useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
 import { Link } from "react-router-dom";
 import PropTypes from "prop-types";
 
 import Paper from "@material-ui/core/Paper";
 import Typography from "@material-ui/core/Typography";
 import DragHandleIcon from "@material-ui/icons/DragHandle";
+import ButtonGroup from "@material-ui/core/ButtonGroup";
+import Button from "@material-ui/core/Button";
+import { makeStyles, useTheme } from "@material-ui/core/styles";
 
+import { ra_to_hours, dec_to_hours } from "../units";
 import * as profileActions from "../ducks/profile";
 import WidgetPrefsDialog from "./WidgetPrefsDialog";
+import { useSourceListStyles } from "./RecentSources";
+import SourceQuickView from "./SourceQuickView";
 
-import styles from "./TopSources.css";
+const useStyles = makeStyles(() => ({
+  header: {},
+  timespanSelect: {
+    display: "flex",
+    width: "100%",
+    justifyContent: "center",
+    marginBottom: "0.5rem",
+    "& .MuiButton-label": {
+      color: "gray",
+    },
+  },
+  sourceListContainer: {
+    height: "calc(100% - 5rem)",
+    overflowY: "scroll",
+    marginTop: "0.625rem",
+    paddingTop: "0.625rem",
+  },
+}));
+
+const getStyles = (timespan, currentTimespan, theme) => ({
+  fontWeight:
+    timespan.label === currentTimespan.label
+      ? theme.typography.fontWeightBold
+      : theme.typography.fontWeightMedium,
+});
+
+const timespans = [
+  { label: "PAST 24 HRS", sinceDaysAgo: "1" },
+  { label: "PAST 7 DAYS", sinceDaysAgo: "7" },
+  { label: "PAST 30 DAYS", sinceDaysAgo: "30" },
```

Comment:
Can we also add 6months and 1yr?

Reply:
Added. Also made option labels more concise and added tooltips for the detailed label:

![Screenshot 2020-09-21 132354](https://user-images.githubusercontent.com/17696889/93818000-a0753900-fc0e-11ea-9ca7-1a872972872d.png)